mbtiles/src/mbtiles.rs

```diff
@@ -55,6 +55,85 @@ pub struct PatchFileInfo {
     pub patch_type: Option<PatchType>,
 }
 
+/// A reference to an MBTiles file providing low-level database operations.
+///
+/// `Mbtiles` represents a reference to an [MBTiles](https://github.com/mapbox/mbtiles-spec)
+/// file without holding an open connection. It provides methods for opening connections
+/// and performing tile operations directly.
+///
+/// # MBTiles Schema Types
+///
+/// MBTiles files can use one of three schema types (see [`MbtType`]):
+/// - [`MbtType::Flat`] - Single table with all tiles, no deduplication
+/// - [`MbtType::FlatWithHash`] - Single table with tiles and MD5 hashes
+/// - [`MbtType::Normalized`] - Separate tables for deduplication via hashing
+///
+/// Use [`detect_type`](Self::detect_type) to determine which schema a file uses.
+///
+/// # Connection Management
+///
+/// `Mbtiles` requires you to manage SQLite connections explicitly. For concurrent
+/// tile serving, consider using [`crate::MbtilesPool`] instead, which provides connection pooling.
+///
+/// # Examples
+///
+/// ## Reading tiles from an existing file
+///
```

Comment:
We likely should warn here of this pitfall 😉

```suggestion
///
/// > [!NOTE]
/// > Note that there are both [osgeos' Tile Map Service](https://wiki.openstreetmap.org/wiki/TMS) and [xyz Slippy map tilenames](https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames) tiling shemes.
/// > They differ only in if the y coordinate direction.
/// > **The default in mapbox and maplibre is xyz.***
/// > **The default in mbtiles is tms.***
///
```